guides/21_Background_Filter_VideoFX_Processor.md

```diff
@@ -75,38 +75,34 @@ The following table lists the browsers and version that support background filte
-
+# Using a Content Security Policy<a id="content-security"></a>
+Modern web applications use Content Security Policy to protect users from certain classes of attacks. An application that utilizes the `VideoFxProcessor` will need to include the following in policy directives to allow the Amazon Chime SDK access to resources it needs at runtime:
+## Required CSP Directives ##
+### Script Policy ###
+To function, the `VideoFxProcessor` class must load JavaScript classes at runtime from an Amazon\-owned CDN\. These classes implement post\-processing for video using WebGL2\. To grant an application access to fetching and running these classes, you must include the following directives:
 ```
-Refused to connect to
-'https://static.sdkassets.chime.aws/bgblur/workers/worker.js…'
-because it violates the document's content security policy.
+script-src       'self' blob: https://*.sdkassets.chime.aws
+script-src-elem  'self' blob: https://*.sdkassets.chime.aws
 ```
-
-The following example shows a typical content security policy\. It includes the tags needed to access the resources needed to run the filters\. You can adapt this example as needed\. 
-
+Please note that `script-src` and `script-src-elem` directives are both required for full support on Safari and Firefox browsers\.
+### Worker Policy ###
+The `VideoFxProcessor` loads JavaScript classes as a blob to run a web worker thread, which processes video using Machine Learning (ML) models\. To grant an application access to fetching and using this worker, include:
+```
+worker-src       'self' blob: https://*.sdkassets.chime.aws
+```
+### WebAssembly Policy ###
+The `VideoFxProcessor` loads a WebAssembly (Wasm) module from the same Amazon-owned CDN\. In Chrome 95 and later, compiled Wasm modules cannot be passed across multiple module boundaries\. To allow fetching and instantiating these modules, please also include `'wasm-unsafe-eval'` to the `script-src` directive\. 
+
+One can read further on the Content Security Policy’s documentation for WebAssembly [here](https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md)\.  Ideally the `wasm-eval` tag would be better used, however it yet to gain official standardization. Correspondence on the directive can be found [here](https://bugs.chromium.org/p/v8/issues/detail?id=7041)\.
+## Optional CSP Directives ##
+### Background Image Policy ###
+To use the background replacement filter with a dynamically loaded background image, `VideoFxProcessor` must be able to access the image\. Include a `connect-src` directive with the domain that hosts the image\.
+## CSP Errors ##
+If any any of the required directives are omitted then the VideoFxProcessor will not be able to properly instantiate and the processor will be unsupported\. Additionally, the following (or similar) errors will likely appear in the browser console:
 ```
-<head>
```

Comment:
Can we keep this section ? It is useful to have a sample CSP setup

Reply:
Sure, I can add it back. But the difficulty will be drawing the line for exactly what the example is going to support. 

If you prefer the csp demo be specifically for the videofxprocessor it would look like this:
```
<meta http-equiv="Content-Security-Policy" 
content= "base-uri         'self';
          script-src       'self' blob: 'wasm-unsafe-eval' https://*.sdkassets.chime.aws;
          script-src-elem  'self' blob: https://*.sdkassets.chime.aws;
          worker-src       'self' blob: https://*.sdkassets.chime.aws;
">
```

But in order to support the actual video/audio that is essential to running a meeting (but not relevant to the videofxprocessor), we would also need to add this line:
```
connect-src      'self' https://*.chime.aws wss://*.chime.aws https://*.amazonaws.com wss://*.chime.aws https://*.ingest.chime.aws;
```

I am fine with either option here. Including the secondary aspect might cause some confusion. But not including it could also lead developers to think the example is an out of box csp configuration for their application. This confusion is what lead me to omit the example all together and alternatively just specify individual directives and their corresponding justification.

Reply:
I think its better to have a minimum set that ppl can just copy and it works even though it may include more thing than just videofxprocessor. You can also make a note that we also include the essential needed for audio video in the sample.

Reply:
Sounds good. Added.